src/faultlog/faultlog_main.cpp

```diff
@@ -25,6 +26,111 @@ using Severity = sdbusplus::xyz::openbmc_project::Logging::server::Entry::Level;
 
 using Binary = std::vector<uint8_t>;
 using PropVariant = sdbusplus::utility::dedup_variant_t<Binary>;
+
+/** @brief Helper method to create faultlog pel
+ *
+ *  @param[in] bus - D-Bus to attach to
+ *  @param[in] guardRecords - hardware isolated records to parse
+ *  @param[in] hostPowerOn - flag to check if called during host IPL
+ */
+void createNagPel(sdbusplus::bus::bus& bus,
+                  const GuardRecords& unresolvedRecords, bool hostPowerOn)
+{
+    int guardCount = GuardWithEidRecords::getCount(unresolvedRecords);
+    int manualGuardCount = GuardWithoutEidRecords::getCount(unresolvedRecords);
+    int unresolvedPelsCount = UnresolvedPELs::getCount(bus, hostPowerOn);
+    int deconfigCount = DeconfigRecords::getCount();
+
+    if ((guardCount > 0) || (manualGuardCount > 0) ||
+        (unresolvedPelsCount > 0) || (deconfigCount > 0))
+    {
+        std::unordered_map<std::string, std::string> data;
+        data.emplace("GUARD_WITH_ASSOC_ERROR_COUNT",
```

Comment:
We could use an initializer list here which would be more efficient and better compiler optimization. Like
`std::unordered_map<std::string, std::string> data = {{"GUARD_WITH_ASSOC_ERROR_COUNT", std::to_string(guardCount)}, {...,...}, ...}`